### PR TITLE
Add 50SAT documentation under the new format

### PR DESCRIPTION
**Pull request by venku122**
This is an example PR for how reviews would be handled. In the diff interface, reviewers can comment directly on the .tex file. Once all comments and issues are resolved, the reviewers can mark the PR as Good, and the user or reviewers could merge it in. This would add a new folder to the root directory with the .tex file, any additional files like images, and the final PDF.

P.S. Is it possible to move the template files to the root directory while still letting LaTEX use them to generate PDFs? Its inflating the number of additions in the PR, which is not really an issue, but could be cleaner.

### CODE REVIEW

File: 50SAT/SPEX$50SAT.tex
Comment (venku122) [inline 50SAT/SPEX$50SAT.tex:99]: @runphilrun This is an example comment. I can provide feedback on a per line basis. There is also a review option that lets a user add a batch of comments at once. The author can reply to this comment directly, and then make the changes locally and push a new commit. Each commit is tracked within one PR.

We can also enable Merge squashing, where all the commits in a branch are combined into a single message when merging to master. This means an individual can make many small commits like `Fix typo`, `add image`, etc, and when looking through the commit history of master, it will look like `Added Project 1`, `Added Project 2`.

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [inline 50SAT/SPEX$50SAT.tex:3]: Help text should probably be deleted before submitting for review. It makes reviewing much harder when you have to scroll through a ton of comment lines.

File: 50SAT/IEEE Journal Template/IEEEtran.cls
Comment (runphilrun) [inline 50SAT/IEEE Journal Template/IEEEtran.cls]: Template files should not be included in PDD folders

**Comment by runphilrun** [comment]
> Is it possible to move the template files to the root directory while still letting LaTEX use them to generate PDFs? Its inflating the number of additions in the PR, which is not really an issue, but could be cleaner.

I think you can do this if you specify relative paths to the class files. Not confirmed but worth a try

**Comment by venku122** [comment]
@runphilrun I was able to build with the IEEE folder and .cls file removed. I guess it just needs a .cls file at the same level as the .tex file?

New commit has the .pdf, .tex, and spexformat.cls

I'm also looking into automatic ways to rip out comments. There is a lot in the template, but you're right, it hurts editability.

**Review by runphilrun — DISMISSED** [review]
I like this structure. Maybe we should have two .tex template documents - one called EditMe.tex that does not contain help comments, and leave the other one HelpMe.tex as is.

**Comment by venku122** [comment]
Maybe, I think it would hurt the writing experience. I know I referred to the comments a lot while writing each section. Task switching to another document might be annoying. Also would that cause compile problems having two .tex files?

**Comment by venku122** [comment]
@runphilrun  Welp, 46a0e5f built without the SPEXformat.cls file. I guess we can drop it?

That brings us down to just the .tex file, pdf, and if the user needs a figures folder or a bibliography.

**Comment by runphilrun** [comment]
Strange. 
* I placed the 50$SAT folder in a directory outside the repository and was able to build the PDF using `\documentclass[conference]{IEEEtran}` without the `IEEEtran.cls` present
* When attempting the same thing with `\documentclass[conference]{SPEXformat}` the build fails unless `SPEXformat.cls` is in the same directory

Maybe TeX Live has IEEEtran built in?

PS There's some styling things that I'm going to submit changes to

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [inline 50SAT/SPEX$50SAT.tex:25]: Why did you add this line? It is unnecessary if you simply add the `img/` to your image file names.

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [inline 50SAT/SPEX$50SAT.tex]: `\usepackage{booktabs}` + https://www.inf.ethz.ch/personal/markusp/teaching/guides/guide-tables.pdf = nicer tables

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [inline 50SAT/SPEX$50SAT.tex]: I think you're using the wrong dash here. It should be an _en_ dash or _em_ dash instead of a hyphen (I forget which). 

LaTeX makes one automatically if you use `--`. 

If you haven't already, install [linter-chktex](https://atom.io/packages/linter-chktex) or a LaTeX linter, it suggests corrections to style errors like this.

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [inline 50SAT/SPEX$50SAT.tex]: This period should end with `\@` after it to be perfectly correct, but here it doesnt really matter since you start a new line. This is another style error caught by  [linter-chktex](https://atom.io/packages/linter-chktex)

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [inline 50SAT/SPEX$50SAT.tex]: Add interword space: `e.g.\`

See this explanation: https://tex.stackexchange.com/questions/2734/taking-unncessary-space-after-e-g-or-i-e

File: 50SAT/SPEX$50SAT.tex
Comment (venku122) [reply 50SAT/SPEX$50SAT.tex]: I'm using VS-code. I'll need to see if there is an equivalent.

File: 50SAT/SPEX$50SAT.tex
Comment (venku122) [reply 50SAT/SPEX$50SAT.tex:25]: This can be more portable in some applications. Its really a matter of preference.

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [reply 50SAT/SPEX$50SAT.tex]: Besides the style guide "violations" (it's not that bad but i can't think of the right word), the table exceeds the column limits.

Consider making the table extend across both columns at the top of the page.

As an example from my CDIM paper:
```tex
\begin{table*}
  \caption{Critical design requirements for the CDIM spacecraft.}
  \small\centering
  \begin{tabular}{@{}llll@{}} \toprule
    Spacecraft Design Driver & Impact & Target \\ \midrule
    Cost & Science capability, instrument architecture & less than \$\SI{1}{B} \\
    Mass & Launch vehicle & less than \SI{1000}{\kilo\gram} \\
    Temperature (OTA) & Passive radiator & \SI{45}{\kelvin} \\
    Temperature (Detector) & Cryocooler & \SI{35}{\kelvin} \\
    Pointing Requirements & Attitude control system & less than \SI{0.5}{arcsec} \\
    Lifetime & Redundancy, RCS propellant & \SI{5}{years} \\
    Orbit & Solar array, thermal management, launch vehicle, telemetry & Sun-Earth \Ltwo{} \\
    \bottomrule
  \end{tabular}
\label{tab:critical-params}
\end{table*}
```

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [reply 50SAT/SPEX$50SAT.tex:3]: ✔️ approve as-is

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [reply 50SAT/SPEX$50SAT.tex:25]: ✔️ approve as-is

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [reply 50SAT/SPEX$50SAT.tex]: 🔧 fix before approval

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [reply 50SAT/SPEX$50SAT.tex]: 🆗 fix preferred but will approve as-is

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [reply 50SAT/SPEX$50SAT.tex]: 🆗 fix preferred but will approve as-is

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [reply 50SAT/SPEX$50SAT.tex]: 🆗 fix preferred but will approve as-is

File: 50SAT/SPEX$50SAT.tex
Comment (venku122) [reply 50SAT/SPEX$50SAT.tex]: How does one break out of the two column layout for the table?

Comment (runphilrun) [reply 50SAT/SPEX$50SAT.tex]: that's what `table*` does, at least in the JAI format class. I'll look into it for IEEE

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [inline 50SAT/SPEX$50SAT.tex]: Multiple issues with this section:
* This bulleted list contains duplicate information from the table above it. 
* "...timeline below:" -> if you use the table, do not reference its location since it is a floating object and position could change. Instead, reference it's Table number. With the `hyperref` package loaded, try `\autoref{tab:timeline}`
* The timeline table does not contain a caption. 
* The timeline table keeps getting pushed to the next page. I think with some format tweaks you could get it to work with single-column, or opt to use the list instead of the table

Comment (venku122) [reply 50SAT/SPEX$50SAT.tex]: 1. The table is designed to replace it. 
2. I think that's a good idea, and solves the inline positioning issue.
3. Does it not? on line `245`?
4. You recommended the table previously when I had the list. I personally prefer the table, once I get it working :p

Comment (runphilrun) [reply 50SAT/SPEX$50SAT.tex]: 1. 👍 
2. 👍 
3. I missed it. Usually I place the caption below the object and I only saw the label. For some reason the caption/table name is not appearing when I build the PDF, Above the table it should say Table #. Estimated Timeline
4. That was before this stupid table was being a bitch haha

Comment (venku122) [reply 50SAT/SPEX$50SAT.tex]: Haha. I'm seeing the caption and table name on my pdf. Do you see it on the github version?
latex_table_50sat

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [inline 50SAT/SPEX$50SAT.tex]: add `\bibliography{SPEX50SAT}` here

File: 50SAT/SPEX$50SAT.tex
Comment (venku122) [reply 50SAT/SPEX$50SAT.tex]: @runphilrun Autoref got me a bright red box in the pdf that links to the table. Is that intended behavior?

Comment (runphilrun) [reply 50SAT/SPEX$50SAT.tex]: 1. I hadn't pulled changes locally. Table looks much better!! We might be able to do something clever to get rid of the "Phase" column. Without the phase column, I think it would fit in a single col

2.  It's a PDF interrnal link, intended. Same thing happens when you make citations. It shouldn't show up in print or in anything other than Adobe Acrobat

File: 50SAT/SPEX$50SAT.tex
Comment (venku122) [reply 50SAT/SPEX$50SAT.tex]: Latest commit has the new bibliography line, but it is not showing up. I think I removed the inline references in a prior version.

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [inline 50SAT/SPEX$50SAT.tex]: This notation is incorrect. The proper way to use `\autoref{}` is to place it in-line as part of the sentence.

For example, 
```tex
\autoref{tab:timeline} shows a preliminary timeline for the project.
```
produces "Table I shows a preliminary timeline for the project." and this reference contains an internal link (if viewing in Acrobat).
Similarly, one may reference figures or even sections using `autoref`, simply replace the argument with the corresponding label (e.g. `\autoref{fig:my-figure}` or `\autoref{sec:my-section}`).

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [inline 50SAT/SPEX$50SAT.tex]: See if you can make the table show up where you want it using placement tags.
```tex
\begin{table*}[htb]
```
More tags and explanation of their usage: https://www.sharelatex.com/learn/Positioning_images_and_tables#!#The_table_environment

This might not work for the `table*` environment, but I know it works well for plain `table`.
Maybe rework your table contents to make them fit better within one column so you don't have to mess with `table*`

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [inline 50SAT/SPEX$50SAT.tex]: * I think the phase column looks best if centered
* Instead of repeating 2, 2, 2, 2 ... I think it might look better and be more readable to put 2 at the first row and then no number for the remaining phase 2 items.
* I think the right column looks better when using "2 weeks or less" rather than "no more than 2 weeks" because it is consistent with the rest of the items having the number appear first
* I've been thinking about the issues with the table spanning 2 rows. I think we can force the table to be the width of one column using the `tabularx` package. I'm not always a fan of using more packages but this one might be pretty useful. more ideas here: https://tex.stackexchange.com/questions/82528/how-to-make-the-table-fit-into-the-column, https://tex.stackexchange.com/questions/312219/autofit-a-table-to-the-column-width

My suggested changes are included below
```tex
\usepackage{tabularx}

\begin{table}[htb]
  \caption{Estimated Timeline}
  \centering
  \begin{tabularx}{\columnwidth}{@{}cXl@{}} \toprule
    Phase & Task & Duration \\ \midrule
    1 & Review existing \$50SAT designs and materials &  2 weeks or less \\
    2 & Subsystem development & 6 weeks \\
      & Order PCB design and/or assembly & 6 weeks \\
      & Review changes to mechanical strucuture and order materials & 2 weeks or less \\
      & Testing of individual subsystems & 2 weeks \\
    3 & System Assembly & 1 week  \\
    4 & System testing & 2 weeks  \\
    5 & Generate documentation and delivery to SPEX & 1 week  \\
    \bottomrule
\end{tabularx}
\label{tab:timeline}
\end{table}
```

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [inline 50SAT/SPEX$50SAT.tex]: Another case of adding `\@` to the end of this sentence. Any sentence that ends in an abbreviation/acronym should end with `.\@`

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [reply 50SAT/SPEX$50SAT.tex]: Yeah you can have an encyclopedia in your `.bib` file but the only things that will show up in your bibliography are the things that are cited.

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [inline 50SAT/SPEX$50SAT.tex:321]: I'm glad you include images in the PDD, but have you thought about including them in other sections like the introduction?

File: 50SAT/SPEX50SAT.bib
Comment (runphilrun) [inline 50SAT/SPEX50SAT.bib:7]: Have they published a paper?

**Review by runphilrun — APPROVED** [review]
No dealbreakers but I think some of the formatting could be improved.

File: 50SAT/SPEX50SAT.bib
Comment (venku122) [reply 50SAT/SPEX50SAT.bib:7]: http://www.50dollarsat.info/home.html

Doesn't look like it, or its locked in a journal.

File: 50SAT/SPEX$50SAT.tex
Comment (venku122) [reply 50SAT/SPEX$50SAT.tex]: 👍

File: 50SAT/SPEX$50SAT.tex
Comment (venku122) [reply 50SAT/SPEX$50SAT.tex]: This worked fantastically.

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [inline 50SAT/SPEX$50SAT.tex:148]: Citation needed. (add `\cite{PocketQube}` to the end of this sentence).

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [inline 50SAT/SPEX$50SAT.tex]: Consider placing the image figures here instead of the appendix

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [inline 50SAT/SPEX$50SAT.tex:248]: Looks much better

**Review by runphilrun — CHANGES_REQUESTED** [review]
Will approve once citation is added

**Comment by runphilrun** [comment]
@venku122 Will approve once citation is added

**Comment by runphilrun** [comment]
@venku122 This is very close to approval, just need to add a citation. Let's get some PDD's merged!

File: 50SAT/SPEX$50SAT.tex
Comment (venku122) [reply 50SAT/SPEX$50SAT.tex:148]: @runphilrun Doesn't seem to work, even with the .bib file

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [inline 50SAT/SPEX$50SAT.tex:318]: Try changing this to 
```latex
\bibliographystyle{IEEEtran}
\bibliography{SPEX50SAT}
```

Also I see a `50SAT` citation in the bib but it is not referenced in the main text

**Review by runphilrun — APPROVED** [review]

File: 50SAT/SPEX$50SAT.tex
Comment (runphilrun) [reply 50SAT/SPEX$50SAT.tex]: Also add ```\bibliographystyle{IEEEtran}``` before ```\bibliography{SPEX50SAT}```

**Pull request merged**
